enclave/src/polkadex_balance_storage.rs

```diff
@@ -150,6 +150,39 @@ pub fn load_balance_storage() -> SgxResult<&'static SgxMutex<PolkadexBalanceStor
     }
 }
 
+// TODO: Write test cases for this function
+pub fn reserve_balance(main_acc: &AccountId, token: AssetId, amount: Balance) -> SgxResult<()> {
+    // Acquire lock on balance_storage
+    let mutex = load_balance_storage()?;
+    let mut balance_storage: SgxMutexGuard<PolkadexBalanceStorage> = mutex.lock().unwrap();
+    let mut balance = balance_storage.read_balance(token.clone(), main_acc.clone())
+        .expect("Unable to read balance from balance storage").clone();
```

Comment:
Is expect okay here, if we are not able to get balance then there is something really wrong with the enclave, so it should be panicking. Thoughts?

Reply:
@brenzi, passing this ball to you

Reply:
I believe it should panic because there is not much we can do without the balance ledger

Reply:
Alright